deals/deals.go

```diff
@@ -92,7 +98,34 @@ func (m *Module) Store(ctx context.Context, waddr string, data io.Reader, dcfgs
 		}
 		proposals = append(proposals, *proposal)
 	}
-	return proposals, failed, nil
+	return dataCid, proposals, failed, nil
+}
+
+func (m *Module) Retrieve(ctx context.Context, waddr string, cid cid.Cid) (io.ReadCloser, error) {
+	f, err := ioutil.TempFile(m.cfg.ImportPath, "retrieve-*")
+	if err != nil {
+		return nil, fmt.Errorf("error when creating tmpfile: %s", err)
+	}
+	addr, err := address.NewFromString(waddr)
+	if err != nil {
+		return nil, err
+	}
+	offers, err := m.api.ClientFindData(ctx, cid)
+	if err != nil {
+		return nil, err
+	}
+	if len(offers) == 0 {
+		return nil, ErrNoOffersAvailable
+	}
+	for _, o := range offers {
+		log.Debugf("trying to retrieve data from %s", o.Miner)
+		if err := m.api.ClientRetrieve(ctx, o.Order(addr), f.Name()); err != nil {
```

Comment:
Will be interesting to see how this should be integrated into a client code. Looks like this is simply writing to a file and blocking until that work is done. If that is the case, maybe we should write the client code in a way that parallelizes calls like this. Just a thought.

Reply:
Actually I think that most of the current impl of `Store` and `Retrieve` will change quite a bit.

I think this code is quite minimal since we hadn't the chance to really test it. Now that we can run more e2e tests, I think we can start making all this code more meaningful. I'm mostly refering to more interesting configurations such as replication.

I still feel there are some gaps in which responsibility this module should have. I hope that clears when we build something that needs this module for its use-case.

Reply:
Sounds good